uaclient/tests/test_cli.py

```diff
@@ -15,6 +20,74 @@
 from uaclient.testing.fakes import FakeConfig
 
 
+BIG_DESC = "123456789 " * 7 + "next line"
+BIG_URL = "http://" + "adsf" * 10
+
+
+class TestCLIParser:
+    @mock.patch("uaclient.cli.entitlements")
+    def test_help_descr_and_url_is_wrapped_at_eighty_chars(
+        self, m_entitlements
+    ):
+        """Help lines are wrapped at 80 chars"""
+
+        def cls_mock_factory(desc, url):
+            return mock.Mock(description=desc, help_doc_url=url)
+
+        m_entitlements.ENTITLEMENT_CLASS_BY_NAME = {
+            "test": cls_mock_factory(BIG_DESC, BIG_URL)
+        }
+
+        parser = get_parser()
+        help_file = io.StringIO()
+        parser.print_help(file=help_file)
+        lines = [
+            " - test: " + " ".join(["123456789"] * 7),
+            "   next line ({url})".format(url=BIG_URL),
+        ]
+        assert "\n".join(lines) in help_file.getvalue()
+
+    def test_help_sourced_dynamically_from_each_entitlement(self):
+        """Help output is sourced from entitlement name and description."""
+        parser = get_parser()
+        cc_eal = ENTITLEMENT_CLASS_BY_NAME["cc-eal"]
+        cis_audit = ENTITLEMENT_CLASS_BY_NAME["cis-audit"]
+        esm = ENTITLEMENT_CLASS_BY_NAME["esm-infra"]
+        fips = ENTITLEMENT_CLASS_BY_NAME["fips"]
+        fips_updates = ENTITLEMENT_CLASS_BY_NAME["fips-updates"]
+        livepatch = ENTITLEMENT_CLASS_BY_NAME["livepatch"]
+        fmt_params = {
+            "cc_desc": cc_eal.description,
```

Comment:
I'm still not a fan of using these attributes. If we accidentally modify one of them (either directly, or via modification of the base class, or the inheritance structure), then we won't catch that in testing. I'd prefer for us to hard-code the expected strings here, so that it forms a complete test of the help output (rather than just a test of the help formatting, which is what this covers now). Does that sounds reasonable?

(To be clear, I agree that it's way clearer using the format parameters!)

Reply:
Let's fully satisfy that need for strict unit test of structured output. The way I had it would only show us errors if we updated descriptions/urls to affect word wrap, which was really all I was interested in testing there. With a static representation of the entire expected string, we'll update this test if we change description, name or help_doc_url. Any maybe that *is* something that we want human review on specifically as we could easily introduce typos that don't make sense on user-facing docs.